operator-sdk: build all binaries

yajo opened the pull request:
###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->
Currently the repo provides 3 binaries. The 2 that were missing are used in upstream docker images. But to use dockerTools instead, these need to be packaged with nix!

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->
@arnarg @moduon MT-1075

arnarg reviewed (approved):
Good catch! LGTM

NickCao commented:
Would you like to resolve the conflicts?

yajo commented:
Thanks, rebased.

NickCao commented on pkgs/development/tools/operator-sdk/default.nix:
```diff
@@ -29,9 +29,14 @@ buildGoModule rec {
     "cmd/operator-sdk"
   ];
 
+  nativeBuildInputs = [ makeWrapper ];
+  buildInputs = [ go ];
+
```
Looks like these lines are redundant? (And causing eval failures)

yajo committed:
operator-sdk: build all binaries

Currently the repo provides 3 binaries. The 2 that were missing are used in upstream docker images. But to use dockerTools instead, these need to be packaged with nix!

@moduon MT-1075

yajo commented:
El dom, 22 de ene de 2023 a las 23:26:15 PM, Nick Cao 
***@***.***> escribió:
> Looks like these lines are redundant? (And causing eval failures)
> 

Oops sorry, I didn't pay enough attention. Fixed now.

NickCao commented:
@ofborg build operator-sdk
